src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -439,15 +676,23 @@ public void stop() throws InterruptedException {
 
   private void onStoragePut(Digest digest) {
     try {
-      backplane.addBlobLocation(digest, config.getPublicName());
+
+      // if the worker is a CAS member, it can send/modify blobs in the backplane.
+      if (!config.getOmitFromCas()) {
+        backplane.addBlobLocation(digest, config.getPublicName());
+      }
     } catch (IOException e) {
       throw Status.fromThrowable(e).asRuntimeException();
     }
   }
 
   private void onStorageExpire(Iterable<Digest> digests) {
     try {
-      backplane.removeBlobsLocation(digests, config.getPublicName());
+
+      // if the worker is a CAS member, it can send/modify blobs in the backplane.
+      if (!config.getOmitFromCas()) {
```

Comment:
This should be a check for boolean state (and we can flip the logic internally to be positive: i.e. `isCasShard`) rather than a dereference of the config. Generally it makes things more testable.